src/evergreen/api.py

```diff
@@ -548,3 +561,24 @@ def clear_caches(self):
         ]
         for fn in cached_functions:
             fn.cache_clear()
+
+
+class RetryingEvergreenApi(EvergreenApi):
+    """An Evergreen Api that retries failed calls."""
+
+    def __init__(self, api_server=DEFAULT_API_SERVER, auth=None, timeout=None):
+        """Create an Evergreen Api object."""
+        super(RetryingEvergreenApi, self).__init__(api_server, auth, timeout)
+
+    @retry(retry=retry_if_exception_type(requests.exceptions.HTTPError),
```

Comment:
fancy!

Reply:
This is using the [tenacity|https://tenacity.readthedocs.io/en/latest/] which we have started using. It is pretty cool.